design/reboot-interface.md

```diff
@@ -89,78 +89,39 @@ section of the BareMetalHost status. This records a time after which the Host
 was last booted using the current image. Processes running prior to this time
 may be assumed to have been stopped.
 
-To provide a declarative interface for performing a reboot, a new
-BareMetalHostRebootRequest CRD will be created. The spec of this custom
-resource will contain only a reference to a BareMetalHost. The status will
-contain timestamps for when the Host was powered off and on again. Once the
-power off time is present in the status, the host has been stopped. Each
-RebootRequest will be owned by the Host it refers to, so that if the Host is
-deleted all associated RebootRequest CRs will be cleaned up.
+A new date-time field, ``lastReboot``, will be added to the ``provisioning``
```

Comment:
As I think you later realised, this wasn't intended as a replacement.
But I think you're right, it's redundant.

Reply:
No! I remember now... since the annotation value comes from the user who's machine may be using a different time or timezone, we cannot trust it as the actual time a reboot was requested.  Hence the desire to record the official system time that the reboot was requested.

This and the expectation that it would help filter out duplicates (both in the host controller and in the fencing logic) should have been included in the PR

Reply:
Ah, so this is more like `lastRebootRequested`. How can we tell when a new one has been added though, unless we also store the list of annotations we've previously seen?

Also I thought we said we were going to store any extra state in annotations? I support keeping the `lastPoweredOn` field because I think it will be useful no matter what user interface we end up adding. But this one is bringing us perilously close to my original implementation that nobody liked.

Reply:
Oh yeah, that should work. I guess I was hung up on the name, because it actually records the earliest reboot request, not the latest one. Looking back at metal3-io/baremetal-operator#239 I had almost this exact thing but called it `pendingRebootSince`.
(The only difference was that I trusted the timestamp provided by the user, so if e.g. the fencing process was delayed and something else rebooted the server in the meantime, then it wouldn't reboot again.)

Reply:
Agreed. I've renamed to ``pendingRebootSince``